### PR TITLE
Enable `run_tests` in Windows CI

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

**Comment by certik** [comment]
TODO:
* [ ] The windows tests are passing, but the tests are actually failing, so that has to be fixed
* [ ] The `\r` should be skipped in the tokenizer, then everything should start working, and the change should be quite minimal

**Comment by certik** [comment]
Let's add a `.gitattributes` file and put there something like:
```
tests/*.py -text
```
which "unsets" the text attribute, which according to https://www.git-scm.com/docs/gitattributes#_text should preserve the line ends.

Then we should keep all the files as they currently are, only using LF. Then let's commit a few files with CRLF to test that CR also works. And then all tests should behave exactly the same on Linux as on Windows.

**Comment by Thirumalai-Shaktivel** [comment]
The Windows CI fails. But, I think it doesn't return a non-zero number, so the CI passes.
I think the issue is related to the threading or parallel running of run_tests.

@certik

**Comment by Thirumalai-Shaktivel** [comment]
```
 --- Logging error ---
[221](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:222)
Traceback (most recent call last):
[222](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:223)
  File "C:\Miniconda\envs\test\lib\logging\__init__.py", line 1103, in emit
[223](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:224)
    stream.write(msg + self.terminator)
[224](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:225)
  File "C:\Miniconda\envs\test\lib\encodings\cp1252.py", line 19, in encode
[225](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:226)
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
[226](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:227)
UnicodeEncodeError: 'charmap' codec can't encode character '\u2713' in position 27: character maps to <undefined>
[227](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:228)
 START TEST:  complex2.py
[228](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:229)
Call stack:
[229](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:230)
 START TEST:  constants1.py
[230](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:231)
  File "D:\a\lpython\lpython\run_tests.py", line 131, in <module>
[231](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:232)
    tester_main("LPython", single_test)
[232](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:233)
  File "D:\a\lpython\lpython\src\libasr\compiler_tester\tester.py", line 398, in tester_main
[233](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:234)
    single_test(test,
[234](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:235)
  File "D:\a\lpython\lpython\run_tests.py", line 71, in single_test
[235](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:236)
    run_test(
[236](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:237)
  File "D:\a\lpython\lpython\src\libasr\compiler_tester\tester.py", line 332, in run_test
[237](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:238)
    log.debug(s + " " + check())
[238](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:239)
Message: 'complex1.py * asr \x1b[32m\x1b[1m\u2713 \x1b[39m\x1b[0m'
[239](https://github.com/lcompilers/lpython/actions/runs/4022128153/jobs/6911599418#step:13:240)
Arguments: ()
```

Can the error be related to the check symbol with color?
complex1.py * asr  ✓ <---

**Comment by Thirumalai-Shaktivel** [comment]
I see, The error was because of this: `✓` symbol.

File: src/libasr/compiler_tester/tester.py [hunk 1]
```diff
@@ -64,11 +64,11 @@ class fg:
 
 
 def color(value):
-    return "\033[" + str(int(value)) + "m"
+    return ""
```
Comment (czgdp1807) [inline src/libasr/compiler_tester/tester.py:67]: Why we have the function if we are just going to return an empty string? Can we just replace, `color(...)` calls with "" instead? Or may be for Windows return an empty string but for other operating systems return `"\033[" + str(int(value)) + "m"`.

Comment (Thirumalai-Shaktivel) [reply src/libasr/compiler_tester/tester.py:67]: Yup, this was used for debugging purposes.
If this method is valid for fixing, then I thought to add an option `--no-color` to print the output without color.

Comment (czgdp1807) [reply src/libasr/compiler_tester/tester.py:67]: > I thought to add an option `--no-color` to print the output without color.

I think its already there? I have used it many times before.

Comment (Thirumalai-Shaktivel) [reply src/libasr/compiler_tester/tester.py:67]: Do you mean, for run_tests?

Comment (czgdp1807) [reply src/libasr/compiler_tester/tester.py:67]: https://github.com/lcompilers/lpython/blob/fa026e1437e2e40e51b302683364c02e603c9d09/src/bin/lpython.cpp#L1409

Comment (Thirumalai-Shaktivel) [reply src/libasr/compiler_tester/tester.py:67]: Yup, that is used for AST and ASR. But, I was thinking to add new option in run_tests. Actually, I had problems with run_tests output color in Windows and here in the CI I had the problem with `check` symbol. So, adding new option would be right fixing this.

**Review by czgdp1807 — APPROVED** [review]

**Review by certik — APPROVED** [review]

**Pull request merged**